### DISCUSSION TITLE
How to build this on M1 Apple - getting error 'sys/auxv.h' file not found

### DISCUSSION THREAD

**Discussion by mathuvenkat** [kubernetes-sigs/node-feature-discovery#803] [Q&A]
node-feature-discovery % uname -a
Darwin 21.3.0 Darwin Kernel Version 21.3.0: Wed Jan  5 21:37:58 PST 2022; root:xnu-8019.80.24~20/RELEASE_ARM64_T6000 x86_64
node-feature-discovery % arch -x86_64 zsh
node-feature-discovery % make build
go build -v -o bin -ldflags "-s -w -X sigs.k8s.io/node-feature-discovery/pkg/version.version=v0.9.0-devel-57-gb4c98bb -X sigs.k8s.io/node-feature-discovery/source.pathPrefix=/" ./cmd/...
sigs.k8s.io/node-feature-discovery/source/internal/cpuidutils
# sigs.k8s.io/node-feature-discovery/source/internal/cpuidutils
source/internal/cpuidutils/cpuid_arm64.go:20:10: fatal error: 'sys/auxv.h' file not found
#include <sys/auxv.h>

**Comment by marquiz** [comment]
Hmm, interesting. What is your use case for this? The built-in feature detection of NFD is basically Linux-only and relies heavily on e.g. Linux sysfs and standard filesystem hierarchy. OTOH, nfd could be used as a general labeling solution: disable all built-in (Linux-specific) feature detection and run your own hooks or sidecars doing the detection 🧐 

Out of curiosity, I quickly created a patch that *should* allow darwin builds: https://github.com/kubernetes-sigs/node-feature-discovery/pull/808

Please try out and report back. I think we could merge that if it is useful to you even if we don't officially support anything else than Linux (yet).

**Reply by mathuvenkat** [reply]
It would be great to have it at-least compile on Mac, even if the service is going to be deployed on Linux. Yes thanks, it worked, will be useful for getting it to compile.

**Reply by marquiz** [reply]
Yeah, sure. Let's get #808 merged (you can comment +1 it)
